lib/dialects/db2/query-generator.js

```diff
@@ -0,0 +1,984 @@
+'use strict';
+
+const _ = require('lodash');
+const Utils = require('../../utils');
+const DataTypes = require('../../data-types');
+const AbstractQueryGenerator = require('../abstract/query-generator');
+const randomBytes = require('crypto').randomBytes;
+const Op = require('../../operators');
+
+/* istanbul ignore next */
+const throwMethodUndefined = function(methodName) {
+  throw new Error(`The method "${  methodName  }" is not defined! Please add it to your sql dialect.`);
+};
+
+class Db2QueryGenerator extends AbstractQueryGenerator {
+  constructor(options) {
+    super(options);
+
+    this.OperatorMap = Object.assign({}, this.OperatorMap, {
+      [Op.regexp]: 'REGEXP_LIKE',
+      [Op.notRegexp]: 'NOT REGEXP_LIKE'
+    });
+    this.autoGenValue = 1;
+  }
+
+  createSchema(schema) {
+    return [
+      'CREATE SCHEMA',
+      this.quoteIdentifier(schema),
+      ';'
+    ].join(' ');
+  }
+
+  dropSchema(schema) {
+    // DROP SCHEMA Can't drop schema if it is not empty.
+    // DROP SCHEMA Can't drop objects belonging to the schema
+    // So, call the admin procedure to drop schema.
+    const query = `CALL SYSPROC.ADMIN_DROP_SCHEMA(${ wrapSingleQuote(schema.trim()) }, NULL, ? , ?)`;
+    const sql = { query };
+    sql.bind = [{ ParamType: 'INOUT', Data: 'ERRORSCHEMA' },
+      { ParamType: 'INOUT', Data: 'ERRORTABLE' }];
+    return sql;
+  }
+
+  showSchemasQuery() {
+    return 'SELECT SCHEMANAME AS "schema_name" FROM SYSCAT.SCHEMATA WHERE ' +
+      "(SCHEMANAME NOT LIKE 'SYS%') AND SCHEMANAME NOT IN ('NULLID', 'SQLJ', 'ERRORSCHEMA')";
+  }
+
+  versionQuery() {
+    return 'select service_level as VERSION from TABLE (sysproc.env_get_inst_info()) as A';
+  }
+
+  /**
+   * Returns an insert into command
+   *
+   * @param {string} table
+   * @param {Object} valueHash       attribute value pairs
+   * @param {Object} modelAttributes
+   * @param {Object} [options]
+   *
+   * @private
+   */
+  insertQuery(table, valueHash, modelAttributes, options) {
```

Comment:
This code looks like an exact copy of `insertQuery` from abstract-query-generator. All other dialect use that code modifying it based on `this._dialect.supports`, please use that approach for query generation.

We usually override query generation method when abstract method and supports can't be used to generate dialect based query. These cases include some format or keywords not supported by other dialects

Reply:
@sushantdhiman Addressed it in latest commit. Thanks.